optuna/logging.py

```diff
@@ -174,7 +174,35 @@ def enable_default_handler() -> None:
 def disable_propagation() -> None:
     """Disable propagation of the library log outputs.
 
-    Note that log propagation is disabled by default.
+    Note that log propagation is disabled by default. You only need to use `disable_propagation`
+    to stop log propagation when you use :func:`~optuna.logging.enable_propogation()`.
+
+    Example:
+
+        Stop propogating logs to the root logger and start logging to :obj:`sys.stderr`.
+
+        .. testsetup::
+
+            def objective(trial):
+                x = trial.suggest_float("x", -100, 100)
+                y = trial.suggest_categorical("y", [-1, 0, 1])
+                return x ** 2 + y
+
+        .. testcode::
+
+            import optuna
+
+            optuna.logging.disable_propagation()  # Stop propogating logs to the root logger.
+            optuna.logging.enable_default_handler()  # Start showing logs in sys.stderr.
```

Comment:
By default, the propagation is disabled and the default handler is enabled. So, I think these lines do not affect the behavior of logging. I guess we need to change the scenario of the code example.

For instance, how about the following?
1. disable the default hander
2. create a logger by `logging.getLogger` and set a hander.
3. enable the log propagation
4. execute `study.optimize` and confirm that log messages are shown
5. disable the log propagation
6. execute `study.optimize` and confirm that no log message are shown